libraries/type-length-value/js/src/tlvState.ts

```diff
@@ -0,0 +1,132 @@
+import { TlvInvalidAccountSizeError, TlvInvalidDiscriminatorError } from './errors.js';
+
+export type TlvNumberSize = 1 | 2 | 4 | 8;
+
+export interface TlvDiscriminator {
+    bytes: Buffer;
+}
+
+export type TlvType = Buffer | Uint8Array | number | bigint | TlvDiscriminator;
```

Comment:
I think we can simplify this and provide good readability by just having one type:

```typescript
type Discriminator = Uint8Array;
```

Since `Buffer` is a subtype of `Uint8Array`, it will work as a `Discriminator`, and whenever someone is using an `SplDiscriminator`, they can just provide `.bytes`.

Then we can forgo the switch case later on and just match against the bytes straight up.

Reply:
What about when we want to use `ExtensionType` from spl-token? In js this is a number (which we would then need to explicitly turn into bytes every time we want to use it).

That's why I added the switch + all the supported types in the first place.

Now that I think of it we could just do that in the `getExtensionData(extension: ExtensionType, tlvData: Buffer): Buffer | null` function

Reply:
Yep! Exactly. It's already quite similar to what you've got